### PR TITLE
feat[onboarding]: Add Transactions to Onboarding

### PR DESCRIPTION
**Pull request by silent1mezzo**
Screen Shot 2020-08-18 at 11 09 00 PM

### CODE REVIEW

**Comment by silent1mezzo** [comment]
Still needs tests for frontend

File: src/sentry/static/sentry/app/components/onboardingWizard/taskConfig.tsx [hunk 1]
```diff
@@ -103,6 +104,37 @@ export function getOnboardingTasks(
...
+    {
+      task: OnboardingTaskKey.FIRST_TRANSACTION,
+      title: t('Enable Performance'),
+      description: t(`See what's slow`),
+      detailedDescription: t(
+        `Add Performance to track latency and bottlenecks within your application.
+        Monitor mission-critical endpoints and set metric alerts to proactively address the worst-performing areas.`
+      ),
+      skippable: true,
+      requisites: [OnboardingTaskKey.FIRST_PROJECT, OnboardingTaskKey.FIRST_EVENT],
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/onboardingWizard/taskConfig.tsx:116]: Wouldn't it be possible to send a transaction before your first error? Should this only depend on the project?

File: src/sentry/static/sentry/app/utils/eventWaiter.tsx [hunk 2]
```diff
@@ -8,8 +8,8 @@ import withApi from 'app/utils/withApi';
 
 const DEFAULT_POLL_INTERVAL = 5000;
 
-const recordAnalyticsFirstEvent = ({organization, project}) =>
-  analytics('onboarding_v2.first_event_recieved', {
+const recordAnalyticsFirstEvent = ({key, organization, project}) =>
+  analytics(`onboarding_v2.${key}`, {
```
[the remark this answers is not in the document]
Comment (silent1mezzo) [reply src/sentry/static/sentry/app/utils/eventWaiter.tsx:12]: @markstory Not yet, was going to reach out to better understand the process prior to deploying.

File: src/sentry/static/sentry/app/components/onboardingWizard/taskConfig.tsx (hunk 1 above)
Comment (silent1mezzo) [reply src/sentry/static/sentry/app/components/onboardingWizard/taskConfig.tsx:116]: It is, I wasn't sure if we wanted to fork the onboarding flow at that point or if we wanted people to continue with adding their first project and event and then going down the road of transactions.

I like the idea of having it along side sending your first error but wasn't sure if that needed to go through a longer review process?

File: src/sentry/static/sentry/app/utils/eventWaiter.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (silent1mezzo) [reply src/sentry/static/sentry/app/utils/eventWaiter.tsx:12]: @markstory https://github.com/getsentry/reload/pull/174

**Comment by silent1mezzo** [comment]
Still need to add a signal/receiver to this and tests.

**Comment by PeloWriter** [comment]
For the card on "Enable Performance", we might consider:

Monitor Performance
See slow faster. Set up Performance Monitoring to see everything from macro-level metrics to micro-level spans. Cross-reference transactions with related issues, customize queries, and monitor mission-critical endpoints using metric alerts.

**Review by markstory — APPROVED** [review]

File: src/sentry/receivers/onboarding.py [hunk 3]
```diff
@@ -174,6 +181,25 @@ def record_first_event(project, event, **kwargs):
...
+        organization_id=project.organization_id,
+        task=OnboardingTask.FIRST_TRANSACTION,
+        status=OnboardingTaskStatus.COMPLETE,
+        date_completed=event.datetime,
+    )
+
+    user = Organization.objects.get(id=project.organization_id).get_default_owner()
+
+    analytics.record(
+        "first_transaction.sent",
```
[the remark this answers is not in the document]
Comment (maheskett) [reply src/sentry/receivers/onboarding.py:195]: late to this but had a couple questions. is first transaction already being recorded [here](https://github.com/getsentry/sentry/blob/master/src/sentry/receivers/transactions.py#L11)?

File: src/sentry/receivers/onboarding.py [hunk 4]
```diff
@@ -112,6 +112,13 @@ def record_raven_installed(project, user, **kwargs):
 
 @first_event_received.connect(weak=False)
 def record_first_event(project, event, **kwargs):
+    if event.get_event_type() == "transaction":
```
Comment (maheskett) [inline src/sentry/receivers/onboarding.py:115]: if the first event was an error, would the first transaction not be recorded?

**Pull request merged**